src/ansys/acp/core/_tree_objects/field_definition.py

```diff
@@ -0,0 +1,148 @@
+# Copyright (C) 2022 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from __future__ import annotations
+
+from collections.abc import Iterable, Sequence
+import typing
+
+from ansys.api.acp.v0 import field_definition_pb2, field_definition_pb2_grpc
+
+from .._utils.property_protocols import ReadWriteProperty
+from ._grpc_helpers.linked_object_list import define_polymorphic_linked_object_list
+from ._grpc_helpers.property_helper import (
+    grpc_data_property,
+    grpc_data_property_read_only,
+    grpc_link_property,
+    mark_grpc_properties,
+)
+from .base import CreatableTreeObject, IdTreeObject
+from .element_set import ElementSet
+from .enums import status_type_from_pb
+from .lookup_table_1d_column import LookUpTable1DColumn
+from .lookup_table_3d_column import LookUpTable3DColumn
+from .modeling_ply import ModelingPly
+from .object_registry import register
+from .oriented_selection_set import OrientedSelectionSet
+
+__all__ = ["FieldDefinition"]
+
+
+_SCOPE_ENTITIES_LINKABLE_TO_FIELD_DEFINITION: typing.TypeAlias = typing.Union[
+    ElementSet,
+    OrientedSelectionSet,
+    ModelingPly,
+]
+
+
+@mark_grpc_properties
+@register
+class FieldDefinition(CreatableTreeObject, IdTreeObject):
+    """Instantiate a Field Definition.
+
+    A field definition is used to configure the state of variable materials.
+    For instance, a Lookup Table can be used to define the distribution of a
+    state of material field (e.g. degradation). The field definition allows
+    to define the material field per element or per ply and element.
+
+    Note
+    ----
+
+    Field definitions are currently only supported through (Py)Mechanical.
+    The direction interface of PyACP to (Py)MADL ignored field definitions.
```

Comment:
I wonder if this renders correctly, or if we should use the ``.. note::`` reStructuredText command instead.

Reply:
It does.